### PR TITLE
feat: start adding node roles to the embedded cluster config crd

### PR DESCRIPTION
**Pull request by ricardomaraschini**
adds node roles to the embedded cluster config crd.

### CODE REVIEW

**Review by chris-sanders — APPROVED** [review]
looks good

File: api/v1beta1/config_types.go [hunk 1]
```diff
@@ -30,8 +30,25 @@ type UnsupportedOverrides struct {
 	K0s string `json:"k0s,omitempty"`
 }
 
+// NodeCount holds a series of rules for a given node role.
+type NodeCount struct {
+	Variable []int `json:"variable,omitempty"`
+	Exact    *int  `json:"exact,omitempty"`
+	Range    []int `json:"range,omitempty"`
+}
```
Comment (laverya) [inline api/v1beta1/config_types.go:38]:
```suggestion
// NodeCount holds a series of rules for a given node role.
type NodeCount struct {
	Variable []int `json:"variable,omitempty"` // a list of allowed sizes
	Exact    *int  `json:"exact,omitempty"`
	Range    []int `json:"range,omitempty"` // if set, must be two numbers, a maximum and minimum
}
```

Comment (laverya) [reply api/v1beta1/config_types.go:38]: I just want comments on this

Comment (laverya) [reply api/v1beta1/config_types.go:38]: OR:
```suggestion
// NodeCount holds a series of rules for a given node role.
type NodeCount struct {
	Variable []int `json:"variable,omitempty"` // a list of allowed node counts
	Min    *int  `json:"exact,omitempty"` // the minimum number of nodes (conflicts with Variable)
	Max    *int `json:"range,omitempty"` // the maximum number of nodes (conflicts with Variable)
}
```

**Review by laverya — APPROVED** [review]

**Pull request merged**